pkgs/applications/networking/taler/default.nix

```diff
@@ -1,57 +1,104 @@
+    '';
+    enableParallelBuilding = true;
+    nativeBuildInputs = [
+      autoreconfHook
+    ];
     buildInputs = [
-      curl
-      jansson
       libgcrypt
       libmicrohttpd
+      jansson
       libsodium
-      libtool
       postgresql
-      # sqlite
+      curl
+      jinja2
+      recutils
+      gettext
+      texinfo # Fix 'makeinfo' is missing on your system.
     ];
     propagatedBuildInputs = [ gnunet' ];
-    patches = [ ./exchange-fix-6665.patch ];
+    configureFlags = [ "--with-gnunet=${gnunet'}" ];
+    checkInputs = [ wget curl ];
+    meta = with lib; {
+      description = ''
+        Taler is an electronic payment system providing the ability to pay
+        anonymously using digital cash.  Taler consists of a network protocol
+        definition (using a RESTful API over HTTP), a Exchange (which creates
+        digital coins), a Wallet (which allows customers to manage, store and
+        spend digital coins), and a Merchant website which allows customers to
+        spend their digital coins.  Naturally, each Merchant is different, but
+        Taler includes code examples to help Merchants integrate Taler as a
+        payment system.
+      '';
+      homepage = "https://taler.net/";
+      license = licenses.agpl3Plus;
+      platforms = platforms.gnu ++ platforms.linux;
+    };
+    # FIXME: at least one test wants to access a postgres db:
+    # `ERROR Database connection to 'postgres:///talercheck' failed: could not connect to server: No such file or directory`
+    doInstallCheck = false;
+    installCheck = ''
+      make check
+    '';
   };
 
-  taler-merchant = mkTaler {
+  taler-merchant = stdenv.mkDerivation rec {
     pname = "taler-merchant";
-    version = "0.8.0";
-    sha256 = "sha256-scrFLXeoQirGqhc+bSQKRl84PfUvjrp1uxF7pfOIB9Q=";
-  } {
-    nativeBuildInputs = [ pkg-config ];
-    buildInputs = taler-exchange.buildInputs ++ [ qrencode taler-exchange ];
+    version = "0.8.3";
+    src = fetchurl {
+      url = "mirror://gnu/gnu/taler/${pname}-${version}.tar.gz";
+      sha256 = "sha256-n7YjJfY8HZevk9o4XbXs6isGFHLmusjCfEb+9maAK0U=";
+    };
+    postPatch = ''
+      find . -name '*.sh' -exec sed -i 's%#!/bin/bash%#!${stdenv.shell}%g' {} \;
+    '';
+    nativeBuildInputs = [ pkg-config autoreconfHook ];
+    buildInputs = taler-exchange.buildInputs ++ [
+      qrencode
+      taler-exchange
+      # for ltdl.h
+      libtool
+    ];
     propagatedBuildInputs = [ gnunet' ];
-    PKG_CONFIG = "${pkg-config}/bin/pkg-config";
+    configureFlags = [
+      "--with-gnunet=${gnunet'}"
+      "--with-exchange=${taler-exchange}"
+  ];
+    preCheck = ''
+      find . -name '*.sh' -exec sed -i -e 's%#!/bin/bash%#!${stdenv.shell}%g' -e 's%#!/usr/bin/env bash%#!${stdenv.shell}%g'{} \;
+    '';
+    doCheck = false; # `make check` is meant to be run after installation
+    checkInputs = [ jq ];
+    doInstallCheck = true;
+    installCheck = ''
+      make check
+    '';
```

Comment:
I think we want to use `´checkTarget`` here.

Reply:
done